model/task/task.go

```diff
@@ -958,13 +972,23 @@ func (t *Task) MarkDependenciesFinished(ctx context.Context, finished bool) erro
 				DependencyTaskIdKey: t.Id,
 			}},
 		},
-		bson.M{
-			"$set": bson.M{bsonutil.GetDottedKeyName(DependsOnKey, "$[elem]", DependencyFinishedKey): finished},
-		},
-		options.Update().SetArrayFilters(options.ArrayFilters{Filters: []interface{}{
-			bson.M{bsonutil.GetDottedKeyName("elem", DependencyTaskIdKey): t.Id},
-		}}),
-	)
+		[]bson.M{
+			bson.M{"$set": bson.M{
+				DependsOnKey: bson.M{
+					"$map": bson.M{"input": "$" + DependsOnKey,
+						"in": bson.M{"$cond": bson.M{
+							"if": bson.M{"$eq": []string{bsonutil.GetDottedKeyName("$$this", DependencyTaskIdKey), t.Id}},
+							"then": bson.M{"$setField": bson.M{
+								"field": DependencyFinishedKey,
+								"input": "$$this",
+								"value": finished,
+							}},
+							"else": "$$this",
+						}},
+					}},
+			}},
+			addDisplayStatusCache,
```

Comment:
Same question about this update as above - if the dependencies are marked as finished, does that actually impact the task's display status? Only `task.UnattainableDependencies`/`task.OverrideDependencies` appear to affect the display status.